src/typescript-worker/node/resolve.ts

```diff
@@ -8,144 +8,179 @@
 // https://github.com/nodejs/node/blob/a9dd03b1ec89a75186f05967fc76ec0704050c36/lib/internal/modules/esm/resolve.js
 // and adapted for use in playground-elements.
 
-/**
- * @param {string} match
- * @param {URL} pjsonUrl
- * @param {boolean} isExports
- * @param {string | URL | undefined} base
- * @returns {void}
- */
-function emitFolderMapDeprecation(match, pjsonUrl, isExports, base) {
-  const pjsonPath = fileURLToPath(pjsonUrl);
+import {fileURLToPath} from './url.js';
+import {
+  InvalidModuleSpecifierError,
+  InvalidPackageConfigError,
+  InvalidPackageTargetError,
+  PackagePathNotExportedError,
+} from './errors.js';
+import type {
+  PackageExports,
+  PackageExportsPathOrConditionMap,
+  PackageExportsTarget,
+  PackageJsonWithExports,
+} from '../util.js';
 
-  if (emittedPackageWarnings.has(pjsonPath + '|' + match)) return;
-  emittedPackageWarnings.add(pjsonPath + '|' + match);
-  process.emitWarning(
+function emitFolderMapDeprecation(
+  match: string,
+  pjsonUrl: URL,
+  isExports: boolean,
+  base: string
+): void {
+  const pjsonPath = fileURLToPath(pjsonUrl);
+  console.warn(
     `Use of deprecated folder mapping "${match}" in the ${
       isExports ? '"exports"' : '"imports"'
     } field module resolution of the package at ${pjsonPath}${
-      base ? ` imported from ${fileURLToPath(base)}` : ''
+      base ? ` imported from ${base}` : ''
     }.\n` +
       `Update this package.json to use a subpath pattern like "${match}*".`,
     'DeprecationWarning',
     'DEP0148'
   );
 }
 
-/**
- * @param {string} specifier
- * @param {URL} packageJSONUrl
- * @param {string | URL | undefined} base
- */
-function throwExportsNotFound(subpath, packageJSONUrl, base) {
-  throw new ERR_PACKAGE_PATH_NOT_EXPORTED(
+function makeExportsNotFoundError(
```

Comment:
Oops, yeah I forgot to change one `throw` to a `return`. Fixed.

The reason I changed from a `throwFooError()` pattern to a `throw makeFooError()` pattern is because TypeScript understands that a `throw` statement stops function execution. That means it will narrow types and won't consider a function that ends with a `throw` to be returning `void`. That helped eliminate the need for a number of casts. (It would be nice if TypeScript noticed that a function always throws, and then applied the same logic where it is invoked!).